pyramid/tier_fs.go

```diff
@@ -228,9 +230,15 @@ func (tfs *TierFS) openFile(fileRef localFileRef, fh *os.File) (*ROFile, error)
 		return nil, fmt.Errorf("file stat: %w", err)
 	}
 
-	// No need to check for the store output here,
-	// we already have the file.
-	tfs.eviction.store(fileRef.fsRelativePath, stat.Size())
+	if !tfs.eviction.store(fileRef.fsRelativePath, stat.Size()) {
+		// This is where we get less strict.
+		// Ideally, newly fetched file will never be rejected by the cache.
+		// But if it did, we prefer to serve the file and delete it.
+		// When the user will close the file, the file will be deleted from the disk too.
+		if err := os.Remove(fileRef.fullPath); err != nil {
```

Comment:
Yay!  I think this makes a *lot* of sense.  E.g. the file might equally well have been accepted and immediately evicted -- this way you get the same end result.